### ISSUE TITLE
Battlefield, castle assault, after boulder impact, the building is demolished only after the smoke animation

### ISSUE THREAD

**Issue by Districh-ru** [ihhub/fheroes2#6222]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

In the castle assault, if catapult attack is successful, after the impact of the boulder, the building is demolished only after the animation of the smoke. And it should be demolished in the middle of the smoke animation.
fheroes2:

https://user-images.githubusercontent.com/113276641/202837831-3e0655cb-8584-47fc-b1e0-53be3d030ba3.mp4

original game:

https://user-images.githubusercontent.com/113276641/202475809-2d5fd120-5693-48e3-8517-6a58399a78b1.mp4

### Save file

[0_9962a_0001.zip](https://github.com/ihhub/fheroes2/files/10048937/0_9962a_0001.zip)

### Additional info

The function that removes the building when it is hit (`SetCastleTargetValue`) is called only after the full animation of the catapult attack (`RedrawActionCatapult`).
The possible solution may be to separate smoke animation into separate two functions: the smoke animation before demolition of the building and after it. (`SetCastleTargetValue` cannot be called from `RedrawActionCatapult` as it is a private function of Arena class)

**Comment by zenseii** [comment]
There's also a problem that there should be two smoke clouds when the draw bridge is struck.

**Comment by felix642** [comment]
Looks like a duplicate of #3242.
